packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -370,6 +536,43 @@ export default {
         events = events.filter((e) => !excludedEvents.includes(e.id));
       }
 
+      if (this.filters.declutter.hideUnlabeled.on) {
+        events = events.filter(
+          (e) =>
+            e.labels.size > 0 || e.codeObject.type !== CodeObjectType.FUNCTION
+        );
+      }
+
+      if (
+        this.filters.declutter.hideElapsedTimeUnder.on &&
+        this.filters.declutter.hideElapsedTimeUnder.time > 0
+      ) {
+        events = events.filter(
+          (e) =>
+            e.returnEvent &&
+            e.returnEvent.elapsedTime &&
+            e.returnEvent.elapsedTime >=
+              this.filters.declutter.hideElapsedTimeUnder.time / 1000
+        );
+      }
+
+      if (
+        this.filters.declutter.hideName.on &&
+        this.filters.declutter.hideName.names.length
+      ) {
+        this.filters.declutter.hideName.names.forEach((fqid) => {
+          const codeObject = classMap.codeObjects.filter(
+            (co) => co.fqid === fqid
+          )[0];
+          if (codeObject) {
+            events = events.filter((e) => !codeObject.allEvents.includes(e));
+          }
+          classMap.codeObjects = classMap.codeObjects.filter(
+            (co) => co !== codeObject
+          );
```

Comment:
You should be able to always filter `codeObjects` at the end of applying event filters. It may be easiest to create a new ClassMap by iterating through the events and merging `[event.codeObject, event.codeObject.ancestors()]` into the code object tree.